src/CModel.cc

```diff
@@ -61,11 +63,18 @@ PTR(afw::detection::Footprint) _mergeFootprints(
         afw::detection::Threshold(0x1, afw::detection::Threshold::BITMASK),
         1 // npixMin
     );
+    afw::geom::Point2I pixel(center);
     if (fpSet.getFootprints()->size() > 1u) {
+        result.setFlag(CModelResult::INCOMPLETE_FIT_REGION, true);
+        for (std::size_t n = 0; n < fpSet.getFootprints()->size(); ++n) {
+            if ((*fpSet.getFootprints())[n]->contains(pixel)) {
```

Comment:
I'm not sure about the type methods for FootprintSet, but if it is in some sort of vector format, would it be possible to iterate from beginning to end of it instead of looping over some index? I.E. I assume that you could start at the pointer position of *fpSet.getFootprints() and step forward, either by incrementing the pointer, or following a vector from start to end. This would save from calling getFootPrints() each iteration of the for loop.